doc/query.md

```diff
@@ -119,6 +172,37 @@ query.put("age", gt12);
 
 See below for supported operators (Selections -> Conditions).
 
+#### Text search
+
+A text clause may be used (if a text index exists) as part of a query to perform full text search term matching, phrase matching, prefix wildcard matching, etc.  A text clause can stand on its own as a query or can be part of a compound query (see below).  Text search supports SQLite Enhanced Query Syntax, see [SQLite full text query][ftsQuery] for more details on syntax that is possible with text search.
+
+##### Restrictions
+
+- Only one text clause per query is permitted.
+- All clauses in a query must be satisfied by an index if that query contains a text search clause.
+
+To find all people named `mike` who `likes to read books` use:
+
+```java 
+// Ensure indexes are created
+// JSON index on "name" field
+im.ensureIndexed(Collections.<Object>singletonList("name"), "name_index");
+// TEXT index on "comment" field
+im.ensureIndexed(Collections.<Object>singletonList("comment"), 
+                 "comment_index", 
+                 "text");
+                 
+// query: { "name": "mike", 
+//          "$text" : { "$search" : "\"likes to read books\"" } }
+Map<String, Object> query = new HashMap<String, Object>();
+Map<String, Object> search = new HashMap<String, Object>();
+search.put("$search", "\"likes to read books\"");
+query.put("name", "mike");
+query.put("$text", search);
+```
```

Comment:
I think the main example shouldn't be a phrase search as this isn't what people typically use. Instead make it something like `likes books`.

Then expand with a few examples for common problems:
- phrase search
- prefix search

Also perhaps note whether one or both tokenisers included by default are case sensitive. The aim is that most people shouldn't have to look at the FTS docs for common use-cases.